### PR TITLE
refactor: reduce same logic codes and add more log

### PR DESCRIPTION
**Pull request by fengmk2**
change test codes to async/await too

##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: lib/init.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+const Watcher = require('./watcher');
+
+module.exports = appOrAgent => {
+  const logger = appOrAgent.coreLogger;
+
+  const watcher = appOrAgent.watcher = appOrAgent.cluster(Watcher)
+    .delegate('watch', 'subscribe')
+    .create(appOrAgent.config)
+    .on('info', logger.info.bind(logger))
```
Comment (dead-horse) [inline lib/init.js]: 可以不用 bind 了吧

`.on('info', (...args) => logger.info(...args))`

Comment (fengmk2) [reply lib/init.js]: 好

**Comment by fengmk2** [comment]
@dead-horse 还是先不改了，要不然外面有人写死了 egg 版本怎么办。

**Comment by fengmk2** [comment]
3.1.0

**Pull request merged**